server/src/main/java/org/elasticsearch/common/util/concurrent/AsyncIOProcessor.java

```diff
@@ -121,6 +123,16 @@ private void processList(List<Tuple<Item, Consumer<Exception>>> candidates) {
         }
     }
 
+    private Consumer<Exception> preserveContext(Consumer<Exception> consumer) {
+        ThreadContext.StoredContext storedContext = threadContext.newStoredContext(false);
```

Comment:
perhaps use `threadContext.newRestorableContext(false)`. It's simpler.

Reply:
Thanks, fixed here: acfa0d6880a2a5a0ba8eacc07a02172459152c92